edge-modules/iotedge-diagnostics-dotnet/src/Program.cs

```diff
@@ -85,52 +73,31 @@ static async Task EdgeAgent(string managementUri)
 
         static async Task Upstream(string hostname, string port, string proxy)
         {
+            var httpClientHandler = new HttpClientHandler();
+            httpClientHandler.ServerCertificateCustomValidationCallback = (message, cert, chain, sslPolicyErrors) => {
+                    return true;   //Is valid
+            };
+
             if (proxy != null)
             {
-                Uri proxyUri = new Uri(proxy);
-                IProxyClient proxyClient = MakeProxy(proxyUri);
+                Environment.SetEnvironmentVariable("https_proxy", proxy);
+            }
 
-                // Setup timeouts
-                proxyClient.ReceiveTimeout = (int)TimeSpan.FromSeconds(60).TotalMilliseconds;
-                proxyClient.SendTimeout = (int)TimeSpan.FromSeconds(60).TotalMilliseconds;
+                var httpClient = new HttpClient(httpClientHandler);
+            var logsUrl = string.Format("https://{0}/devices/0000/modules", hostname);
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, logsUrl);
+            HttpResponseMessage httpResponseMessage = await httpClient.SendAsync(httpRequest, HttpCompletionOption.ResponseHeadersRead);
 
-                // Get TcpClient to futher work
-                var client = proxyClient.CreateConnection(hostname, int.Parse(port));
-                client.GetStream();
-            }
-            else
+            var keys = httpResponseMessage.Headers.GetValues("iothub-errorcode");
+            if (keys.Contains("InvalidProtocolVersion") == false)
             {
-                TcpClient client = new TcpClient();
-                await client.ConnectAsync(hostname, int.Parse(port));
-                client.GetStream();
+                throw new Exception($"Got bad response");
```

Comment:
It might be good to be more specific here. Something like "The client did not receive the expected response from hub

Reply:
Fixed